frontend/src/lib/components/BranchButton.svelte

```diff
@@ -0,0 +1,200 @@
+<!-- BranchButton.svelte -->
+<script lang="ts">
+	import { branchName } from '$lib/main';
+	import { derived } from 'svelte/store';
+	import { onMount } from 'svelte';
+	import { apiAddress } from '$lib/net';
+
+	export let initialBranchName: string;
+	let showMenu = false;
+	let existingBranches: string[] = [];
+	let newBranchName: string = '';
+
+	const currentBranch = derived(branchName, ($branchName) => $branchName || initialBranchName);
+
+	async function fetchExistingBranches() {
+		const response = await fetch(`${apiAddress}/api/branches`, {
+			method: 'GET',
+			credentials: 'include',
+			headers: {
+				'Content-Type': 'application/json'
+			}
+		});
+
+		// Check if the response is successful
+		if (!response.ok) {
+			const errorMessage = await response.json();
+			throw new Error(
```

Comment:
If a function is explicitly fallible, it should be documented in the function docstring, explaining what error it throws.

I'm partial to errors as values, and so I'd make the function return `null || Error` instead of throwing an error, but that's not idiomatic so choose whichever you please

Reply:
I think I have resolved this one